osm2pgsql-replication

```diff
@@ -0,0 +1,338 @@
+#!/usr/bin/env python3
+
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# This file is part of osm2pgsql (https://osm2pgsql.org/).
+#
+# Copyright (C) 2006-2021 by the osm2pgsql developer community.
+# For a full list of authors see the git log.
+
+"""
+Update an osm2pgsql database with changes from a OSM replication server.
+
+This tool initialises the updating process by looking at the import file
+or the newest object in the database. The state is then saved in a table
+in the database. Subsequent runs download newly available data and apply
+it to the database.
+
+See the help of the 'init' and 'update' command for more information on
+how to use %(prog)s.
+"""
+
+from argparse import ArgumentParser, RawDescriptionHelpFormatter
+import datetime as dt
+import logging
+import re
+import sys
+import subprocess
+import tempfile
+import traceback
+from pathlib import Path
+import urllib.request as urlrequest
+
+import psycopg2
+
+from osmium.replication.server import ReplicationServer
+from osmium.replication.utils import get_replication_header
+from osmium import WriteHandler
+
+LOG = logging.getLogger()
+
+def connect(args):
+    """ Create a connection from the given command line arguments.
+    """
+    return psycopg2.connect(dbname=args.database, user=args.username,
+                            host=args.host, port=args.port)
+
+
+def compute_database_date(conn, prefix):
+    """ Determine the date of the database from the newest object in the
+        data base.
+    """
+    # First, find the way with the highest ID in the database
+    # Using nodes would be more reliable but those are not cached by osm2pgsql.
+    with conn.cursor() as cur:
+        cur.execute("SELECT max(id) FROM {}_ways".format(prefix))
+        osmid = cur.fetchone()[0] if cur.rowcount == 1 else None
+
+        if osmid is None:
+            LOG.fatal("No data found in the database.")
+            return None
+
+    LOG.info("Using way id %d for timestamp lookup", osmid)
+    # Get the node from the API to find the timestamp when it was created.
+    node_url = 'https://www.openstreetmap.org/api/0.6/way/{}/1'.format(osmid)
+    headers = {"User-Agent" : "osm2pgsql-update"}
```

Comment:
"Get the node" and node_url still seem to refer to a previous version that was based on nodes?